src/redux/saga/facebookSignInSaga.ts

```diff
@@ -0,0 +1,31 @@
+import { AccessToken, LoginManager } from 'react-native-fbsdk-next';
+import auth from '@react-native-firebase/auth';
+import { AccessTokenMap } from 'react-native-fbsdk-next/src/FBAccessToken';
+import { createErrorAlertMessageHelper } from '../../helpers/createErrorAlertMessageHelper';
+import { put } from 'redux-saga/effects';
+import { checkFirebaseUsersSagaAction } from '../actions/checkFirebaseUsersSagaAction';
+
+export function* facebookSignInSaga() {
+  try {
+    // Attempt login with permissions
+    yield LoginManager.logInWithPermissions(['public_profile', 'email']);
+
+    // Once signed in, get the users AccessToken
+    const dataToken: AccessTokenMap = yield AccessToken.getCurrentAccessToken();
+    const accessToken = dataToken.accessToken;
+
+    // Create a Firebase credential with the AccessToken
+    const facebookCredential =
+      auth.FacebookAuthProvider.credential(accessToken);
+
+    // Sign-in the user with the credential
+    yield auth().signInWithCredential(facebookCredential);
+    const { user } = yield auth().signInWithCredential(facebookCredential);
+
+    const userToken = user.uid;
+
+    yield put(checkFirebaseUsersSagaAction({ user, userToken }));
```

Comment:
this is action creator
but @AndrewSakovich, make sense just call saga instead of dispatch action that is not handled in reducer and is not called from components

Reply:
corrected